### ISSUE TITLE
Transactions not disabled, with EnableTracing false, when trace header present in request

### ISSUE THREAD

**Issue by shouryaps** [getsentry/sentry-go#594]
## Summary
When the `Sentry-Trace` header is present in the request with a valid trace id, and `sentry.Init` has `EnableTracing` value as false, transactions are still getting reported. 

Here's the detailed timeline of what happened:
1. Got an email alert from sentry.io on transactions quota usage alert for my project while I have set `EnableTracing` as false, as we are using sentry.io only for error tracking
2. On investigating, found that transactions were being reported only for 3 APIs, while the project has more APIs.
3. On checking further, found that for these 3 APIs, I was getting `Sentry-Trace` header in the request from the external caller (this API is consumed by third parties, and it seems they are using sentry, and this header is getting attached).
4. On removing this header from our reverse proxy, transactions usage again dropped to 0

Found a similar issue with the sentry ruby client as well: https://github.com/getsentry/sentry-ruby/issues/1307.

## Steps To Reproduce
1. I am using sentry as a chi middleware, same as [here](https://gist.github.com/rhcarvalho/66130d1252d4a7b1fbaeacfe3687eaf3)
2. Have set `EnableTracing` as `false` explicitly in `sentry.Init` (although its default false)
3. Now when trying to hit an API with `Sentry-Trace` header present with a valid trace id, transactions appear in the sentry dashboard

## Expected Behavior
With `EnableTracing` set as false (default behavior), `Sentry-Trace` should not lead to transactions usage

### SDK
 - `sentry-go` version: v0.18.0
 - Go version: 1.19
 - Using Go Modules? yes

### Sentry
 - Using hosted Sentry in sentry.io? yes

**Comment by cleptric** [comment]
The linked chi snippet does not seem to start any transactions.
Looking at https://github.com/getsentry/sentry-go/blob/master/tracing.go#L387-L391, we do set `Sampled: false`, if `EnableTracing: false`.

Is there maybe some other code in your app that starts a transaction?

**Comment by shouryaps** [comment]
No, there is no other code, that starts the transaction in the project explicitly.

Checking back on the original issue, I noticed that it was coming in go client **0.14.0**, with `TracesSampleRate` set as 0. Apologies for stating the version as 0.18.0 in the original issue message. Adding the header back from the reverse proxy, I'm able to see transactions dropping with 0.18.0 when the header is passed.

I tried a simple program from [above-stated snippet](https://gist.github.com/rhcarvalho/66130d1252d4a7b1fbaeacfe3687eaf3) and made a GET API to try to replicate and found the issue to be replicating in sentry go client 0.14.0, with `Sentry-Trace` header passed.
This was happening in that version because it was returning in this line itself: https://github.com/getsentry/sentry-go/blob/e7611f7634cbe46c633bc0a0b412f1ef6c355445/tracing.go#L282 while in the new version, there is a check of `EnableTracing` explicitly before this.

Since the issue is not replicating in the latest go client version 0.18.0, I'm closing this issue.

Thanks
